client/src/Main.elm

```diff
@@ -1,5 +1,56 @@
-module Main exposing (main)
+port module Main exposing (Msg(..), main, update, view)
 
-import Html exposing (text)
+import Browser
+import Html exposing (Html, button, div, text)
+import Html.Events exposing (onClick)
+import Json.Encode exposing (Value, string)
 
-main = text "hi"
\ No newline at end of file
+
+port sendMessage : Value -> Cmd msg
+
+
+main =
+    Browser.element
+        { init = init
+        , view = view
+        , update = update
+        , subscriptions = subscriptions
+        }
+
+
+type Msg
+    = Increment
+    | Decrement
+
+
+type alias Model =
```

Comment:
that is the magic type that encapsulate the state right

Reply:
Yes